types/src/checksummed_hex.rs

```diff
@@ -53,33 +57,35 @@ const HEX_CHARS: [char; 22] = [
 ///
 /// [1]: https://eips.ethereum.org/EIPS/eip-55
 pub fn encode(input: &(impl AsRef<[u8]> + ?Sized)) -> String {
-    let input_bytes = input.as_ref();
-    let mut hash_bits = bytes_to_bits_cycle(blake2b_hash(input));
-    let mut hex_output_string = String::with_capacity(input_bytes.len() * 2);
-
-    for mut nibble in bytes_to_nibbles(input_bytes) {
-        if nibble >= 10 && hash_bits.next().unwrap_or(true) {
-            nibble += 6;
-        }
-        hex_output_string.push(HEX_CHARS[nibble as usize])
-    }
-    hex_output_string
+    encode_iter(input).collect()
 }
 
 /// `encode` but it returns an iterator.
```

Comment:
So this automatically checksums only for small inputs now, and otherwise outputs lowercase? That feels like somewhat surprising behavior. At least it should be explained in the documentation.

Reply:
@Fraser999 wanted this behavior for `bytesrepr::Bytes` so for the sake of consistency we're doing it everywhere now.